[
    {
        "content": "<p>The PR <a href=\"https://github.com/leanprover-community/mathlib4/pull/9879\">#9879</a> was delegated to me and after adapting the last suggestion by <span class=\"user-mention\" data-user-id=\"459699\">@JoÃ«l Riou</span>, I waited for CI to pass (which it did) and then commented <code>bors r+</code>. Now the bors build is failing with <code>Mathlib/CategoryTheory/Limits/FilteredColimitCommutesFiniteLimit.lean:407:18: error: Declaration Field is not allowed to be imported by this file.</code> This error did not appear in the previous CI builds. Is this related to the changes in <a href=\"https://github.com/leanprover-community/mathlib4/pull/9879\">#9879</a>? Maybe the import that I added to the <code>FinCategory</code> file?</p>",
        "id": 417619425,
        "sender_full_name": "Christian Merten",
        "timestamp": 1706093294
    },
    {
        "content": "<p>And, why did that not cause failure in regular CI builds?</p>",
        "id": 417619662,
        "sender_full_name": "Christian Merten",
        "timestamp": 1706093416
    },
    {
        "content": "<p>I traced it back: I think it is indeed the line <code>import Mathlib.CategoryTheory.SingleObj</code> in the <code>FinCategory</code> file  because the former transitively imports <code>Mathlib.Algebra.Field.Defs</code> which causes this error.</p>",
        "id": 417620480,
        "sender_full_name": "Christian Merten",
        "timestamp": 1706093727
    },
    {
        "content": "<p>It looks like you need to get the latest master and merge it into that branch. The error will then surface for you locally and you can fix it. And then you can send it back to bors.</p>",
        "id": 417620776,
        "sender_full_name": "Oliver Nash",
        "timestamp": 1706093822
    },
    {
        "content": "<p>Possibly the solution is to split up SingleObj into a part with just the basic definitions, and another part that gives the connections with various algebraic structures.</p>",
        "id": 418009245,
        "sender_full_name": "Kim Morrison",
        "timestamp": 1706158593
    },
    {
        "content": "<p>Another approach would be to create a new file importing both FinCategory and SingleObj.</p>",
        "id": 418009287,
        "sender_full_name": "Kim Morrison",
        "timestamp": 1706158631
    },
    {
        "content": "<p>(I just added this assert_not_exists this week. Apologies that it had tripped you up, but good to know it is doing its job. :-)</p>",
        "id": 418009386,
        "sender_full_name": "Kim Morrison",
        "timestamp": 1706158691
    },
    {
        "content": "<p><span class=\"user-mention silent\" data-user-id=\"110087\">Scott Morrison</span> <a href=\"#narrow/stream/287929-mathlib4/topic/Build.20failing.3A.20declaration.20not.20allowed.20to.20be.20imported/near/418009245\">said</a>:</p>\n<blockquote>\n<p>Possibly the solution is to split up SingleObj into a part with just the basic definitions, and another part that gives the connections with various algebraic structures.</p>\n</blockquote>\n<p>I just went with moving the instance from the <code>FinCategory</code> file to <code>SingleObj</code> and importing <code>FinCategory</code> there, this fixed the <code>assert_not_exists</code> error.</p>",
        "id": 418147206,
        "sender_full_name": "Christian Merten",
        "timestamp": 1706209819
    },
    {
        "content": "<p>The only \"algebraic\" import in <code>SingleObj</code> is <code>Mathlib.Algebra.Category.MonCat.Basic</code> which is necessary for the definition, so I think splitting up <code>SingleObj</code> wont help here.</p>",
        "id": 418147334,
        "sender_full_name": "Christian Merten",
        "timestamp": 1706209879
    }
]